### DISCUSSION TITLE
new version of fake sensors

### DISCUSSION THREAD

**Discussion by jho59** [cpetrich/counterfeit_DS18B20#42] [General]
Hi,

of course I know, that 20 sensors for 5 Euro can't be original. What I got from a chinese platform:
1. Front side for TO92 case is lasered, but some lines are dots with gaps.
2. 64bit IDs are  of correct pattern.
3. Can switch between 9, 10, 11, 12bit resolution. Values increments of 0.5°C, 0.25°C etc.
4. Conversion time is 2 ms - yes TWO milliseconds. The sensors don't recognize value of waitForConversion.

Bye, Jürgen

**Comment by cpetrich** [comment]
Thank you for the pointer.
Just had a look, seems like the number of distributors (manufacturers)? is proliferating.
Planning on following up on this.
Can you share a ROM and the marking on the case?

**Reply by jho59** [reply]
Hi,

1. The marking is:
DALLAS
18B20
2402C4
+817AB
On backside a dent, but no character P.

2. Don't know, what you mean with "share a ROM". I let discover_fake_DS18B20.ino run:

28-3E-43-87-00-00-00-18: ROM ok.
  Scratchpad Register: 90/01/7F/80/7F/FF/10/10/1D
  Info only: Scratchpad bytes 2,3,4 (7F/80/7F):  not Maxim default values 4B/46/7F.
  Scratchpad byte 5 (0xFF):  ok.
  Scratchpad byte 6 (0x10):  ok.
  Scratchpad byte 7 (0x10):  ok.
  0x4E modifies alarm registers:  ok.
  0x4E accepts 10 bit resolution:  ok.
  0x4E preserves reserved bytes:  ok.
  0x4E accepts 12 bit resolution:  ok.
  0x4E preserves reserved bytes:  ok.
  Checking byte 6 upon temperature change: not necessary. Skipped.
  --> Sensor responded like a genuie Maxim.
      Not tested: EEPROM, Parasite Power, and undocumented commands.

**Reply by uzi18** [reply]
Please try to check parasite mode - fakes got problem with report power status

**Reply by jho59** [reply]
1. Sensors don't send 0 as long as conversion is ongoing and 1 when finished. So they are useless in my opinion. So why should I spend more time to try out parasite-powered mode?
2. Datasheet shows a MOSFET, but make no suggestion about type. At the moment I don't have anything for Vdd = 3,3V.

**Comment by cpetrich** [comment]
I think I managed to purchase these chips from a different supplier (mine were not marked DALLAS). I added them as a new Family H to the list. I still have to update the sketches and some parts of the summary to account for the new Families.

Again, I much appreciate your pointer.

**Reply by cpetrich** [reply]
rebranded as Family A3

**Comment by afallows** [comment]
Thanks for developing this!

Just had a batch of 5 stainless encapsulated probes show up as this A3 type as well.

28-34-0F-86-00-00-00-90: ROM pattern pre-dates C4 die version, suggesting the chip is either over 15 years old or a clone.
                         Assuming sensor is a clone. Error.
  Scratchpad Register: 74/01/7F/80/7F/FF/0C/10/DA
  Info only: Sensor is not reporting Parasite Power Mode. Ok.
  Info only: Scratchpad bytes 2,3,4 (7F/80/7F): not DS18B20 default values 4B/46/7F.
  Scratchpad <byte 5> = 0xFF: ok.
  Scratchpad <byte 6> = 0x0C: ok.
  Scratchpad <byte 7> = 0x10: ok.
  Last three bytes of scratchpad after sending 5 bytes: FF/0C/10 ok.
  Scratchpad config register <byte 4> after writing 0x00: 0x1F ok.
  Scratchpad config register <byte 4> after writing 0xFF: 0x7F ok.
  0x4E modifies alarm registers: ok.
  0x4E accepts 10 bit resolution: ok.
  0x4E preserves reserved bytes: ok.
  0x4E accepts 12 bit resolution: ok.
  0x4E preserves reserved bytes: ok.
  Checking <byte 6> upon temperature change: ok.
    Info only: Temperature reading is 23.50 oC.
  Testing if end of temperature conversion can be polled: ok.
    Polling status valid immediately: Sensor failed to signal immediately 4 out of 4 times. Error.
    Conversion speed at different resolutions: 9-bit: 77 ms, 10-bit: 154 ms, 11-bit: 309 ms, 12-bit: 617 ms.
      Changing resolution possible. Ok.
      Conversion time doubles with each bit in resolution. Ok.
      12-bit conversion time as expected. Ok.
  --> Sensor is not a genuine DS18B20 based on 2 deviations.

28-45-EA-85-00-00-00-BF: ROM pattern pre-dates C4 die version, suggesting the chip is either over 15 years old or a clone.
                         Assuming sensor is a clone. Error.
  Scratchpad Register: 75/01/7F/80/7F/FF/0B/10/F7
  Info only: Sensor is not reporting Parasite Power Mode. Ok.
  Info only: Scratchpad bytes 2,3,4 (7F/80/7F): not DS18B20 default values 4B/46/7F.
  Scratchpad <byte 5> = 0xFF: ok.
  Scratchpad <byte 6> = 0x0B: ok.
  Scratchpad <byte 7> = 0x10: ok.
  Last three bytes of scratchpad after sending 5 bytes: FF/0B/10 ok.
  Scratchpad config register <byte 4> after writing 0x00: 0x1F ok.
  Scratchpad config register <byte 4> after writing 0xFF: 0x7F ok.
  0x4E modifies alarm registers: ok.
  0x4E accepts 10 bit resolution: ok.
  0x4E preserves reserved bytes: ok.
  0x4E accepts 12 bit resolution: ok.
  0x4E preserves reserved bytes: ok.
  Checking <byte 6> upon temperature change: ok.
    Info only: Temperature reading is 23.62 oC.
  Testing if end of temperature conversion can be polled: ok.
    Polling status valid immediately: Sensor failed to signal immediately 4 out of 4 times. Error.
    Conversion speed at different resolutions: 9-bit: 77 ms, 10-bit: 155 ms, 11-bit: 309 ms, 12-bit: 617 ms.
      Changing resolution possible. Ok.
      Conversion time doubles with each bit in resolution. Ok.
      12-bit conversion time as expected. Ok.
  --> Sensor is not a genuine DS18B20 based on 2 deviations.

Your choice> 3

28-34-0F-86-00-00-00-90: Family A3 (Clone).
28-45-EA-85-00-00-00-BF: Family A3 (Clone).
